HTTP Server.core/Server.cs

```diff
@@ -10,10 +11,12 @@ public class Server
         public bool Running = false;
         private ISocket socket;
         private ISocket clientConnection;
+        private string directoryPath;
 
-        public Server Start(int port, ISocket startSocket)
+        public Server Start(int port, ISocket startSocket, string directory)
```

Comment:
What if there's another option you need to pass to the server in the future? Can you structure the code such that the server constructor doesn't need to change when it needs more configuration?